specification/ai/Azure.AI.Unified/datasets/routes.tsp

```diff
@@ -0,0 +1,136 @@
+import "@typespec/rest";
+import "@azure-tools/typespec-autorest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "../common/models.tsp";
+import "./models.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+using Azure.Core.Foundations;
+
+namespace Azure.AI.Unified;
+
+interface Datasets {
+  #suppress "@azure-tools/typespec-azure-core/operation-missing-api-version"
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations"
+  @doc("List latest version of each dataset in a project.")
+  @route("/datasets")
+  @get
+  listLatestDatasets(
+    @doc("Continuation token for pagination. This is the nextLink from the previous response.")
+    @query("$continuationToken")
+    $continuationToken?: string,
+
+    @doc("View type for including/excluding (for example) archived entities.")
+    @query("listViewType")
+    listViewType?: ListViewType,
+  ): PagedDatasetVersion;
+
+  #suppress "@azure-tools/typespec-azure-core/operation-missing-api-version"
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations"
+  @doc("List dataset versions of a specific dataset")
+  @route("/datasets/{name}/versions")
+  @get
+  listDatasetVersions(
+    @doc("Container name.")
+    @pattern("^[a-zA-Z0-9][a-zA-Z0-9\\-_]{0,254}$")
+    @path
+    name: string,
+
+    @doc("Please choose OrderBy value from ['createdtime', 'modifiedtime']")
+    @query("$orderBy")
+    $orderBy?: string,
```

Comment:
This should be a union of string and other pre-defined values, defined in the \common\Models.tsp files. It is also used by indexes